chromium_src/chrome/browser/profiles/gaia_info_update_service_factory.cc

```diff
@@ -0,0 +1,10 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/profiles/brave_gaia_info_update_service.h"
+
+#define  GAIAInfoUpdateService BraveGAIAInfoUpdateService
```

Comment:
I am getting an error in Debug build on Windows:

<pre>
lld-link: error: undefined symbol: public: static class GAIAInfoUpdateService * __cdecl GAIAInfoUpdateServiceFactory::GetForProfile(class Profile *)
>>> referenced by C:\bb4\brave-browser\src\chrome\browser\profiles\profile_impl.cc:671
</pre>

This changes the return type of `GAIAInfoUpdateService* GAIAInfoUpdateServiceFactory::GetForProfile` as well.

Reply:
It might be easier to override `ShouldUseGAIAProfileInfo` directly by adding chromium_src override of `gaia_info_update_service.h` like so:
<pre>
#ifndef BRAVE_CHROMIUM_SRC_CHROME_BROWSER_PROFILES_GAIA_INFO_UPDATE_SERVICE_H_
#define BRAVE_CHROMIUM_SRC_CHROME_BROWSER_PROFILES_GAIA_INFO_UPDATE_SERVICE_H_

#define ShouldUseGAIAProfileInfo                           \
  ShouldUseGAIAProfileInfo_ChromiumImpl(Profile* profile); \
  static bool ShouldUseGAIAProfileInfo
#include "../../../../../chrome/browser/profiles/gaia_info_update_service.h"
#undef ShouldUseGAIAProfileInfo

#endif  // BRAVE_CHROMIUM_SRC_CHROME_BROWSER_PROFILES_GAIA_INFO_UPDATE_SERVICE_H_
</pre>

and changing the `gaia_info_update_service.cc` override to be:
<pre>
#include "chrome/browser/profiles/gaia_info_update_service.h"

#define ShouldUseGAIAProfileInfo ShouldUseGAIAProfileInfo_ChromiumImpl
#include "../../../../../chrome/browser/profiles/gaia_info_update_service.cc"
#undef ShouldUseGAIAProfileInfo

// static
bool GAIAInfoUpdateService::ShouldUseGAIAProfileInfo(Profile* profile) {
  return false;
}
</pre>

and then there will be no need for `browser/profiles/brave_gaia_info_update_service.{h,cc} or  `chromium_src/chrome/browser/profiles/gaia_info_update_service_factory.cc` overrides.

Reply:
this is helpful, thanks @mkarolin. I'll update